### ISSUE TITLE
Regressors retained when t.pval=0

### ISSUE THREAD

**Issue by gsucarrat** [gsucarrat/gets#67]
When `t.pval=0`, then no regressors should be retained. However, this is not always the case, as the following code illustrates:

`set.seed(1)
x1 <- rnorm(100)
x2 <- rnorm(100)
x3 <- rnorm(100)
x4 <- rnorm(100)
y <- 8+2*x1+3*x2+4*x3+rnorm(100)
getsFun(y, cbind(1,x1,x2,x3,x4), t.pval=0) #erroneously retains 1 and 4`

Thanks to Santiago Diaz Espitia for notifying us about the issue. The problem may affect all functions that rely on `getsFun()`. This includes, amongst other, `gets.arx()`, `gets.logitx()`, `gets.lm()` and `isat.default()`.

**Comment by SantiagoD999** [comment]
It works perfectly with the getsm function:

set.seed(1)
x1<-rnorm(100)
x2<-rnorm(100)
x3<-rnorm(100)
x4<-rnorm(100)
y<-8+2*x1+3*x2+4*x3+rnorm(100)
Y<-zoo(cbind(y,x1,x2,x3,x4))

getsm(arx(Y[,1],mxreg=Y[,-1]),t.pval=0,do.pet = FALSE)

**Comment by jkurle** [comment]
Hi Santiago and thank you for raising this issue. You probably already delivered the solution yourself.

In your original code and Genaro's example, you do the PET. While in your comment above, you turn PET off. If we modify Genaro's code to also turn PET off, we find the expected result that the terminal model is empty.

```
set.seed(1)
x1 <- rnorm(100)
x2 <- rnorm(100)
x3 <- rnorm(100)
x4 <- rnorm(100)
y <- 8+2*x1+3*x2+4*x3+rnorm(100)
getsFun(y, cbind(1,x1,x2,x3,x4), t.pval=0, do.pet=FALSE) # best terminal model is empty, as expected
```

Now, what is going on here? When setting `t.pval = 0` without specifying `wald.pval` separately, then also `wald.pval = 0` by default. So we would expect that the PET never rejects. However, at least in this example and potentially other cases, the PET yields test statistics that are huge (well above 4500), which is so large that R actually returns a p-value of 0 for this. Therefore, the deleted regressor is added as a `keep` variable and we end up with terminal models that are not empty. See code snippet below:

https://github.com/gsucarrat/gets/blob/85b73e62e24520548893a6fa15b3bbe4f1fc8824/gets/R/gets-base-source.R#L1613-L1632

So basically, this behaviour is due to rounding/machine imprecision.

How do we deal with this? I'm not sure how relevant this case is in practice since users can set `do.pet=FALSE` if they want to use `t.pval=0`. This still does not guarantee that we delete all regressors because in theory, we can also get t-statistics for individual regressors that are so big that their p-value is returned as exactly 0, again due to rounding. @SantiagoD999 Do you mind explaining your use case of this setting as a guidance for how to deal with this?

**Comment by SantiagoD999** [comment]
It worked perfectly, thank you very much.

**Comment by jkurle** [comment]
Glad to hear!

@gsucarrat: How do we deal with the `include.gum` argument being ignored/deprecated? Apparently, it was last modified (added?) in the commit 6f424a50b42e9c4c336a82cd08ef9ef167dff3c9 on 2/10/2020.

**Comment by gsucarrat** [comment]
First of all Jonas, thanks a lot! As for `include.gum`, I think I was the one who deprecated the argument, but I do not exactly remember why... We could try to simply re-enable the argument in `isat()` and see what happens. If something breaks, then maybe an idea could be to handle the situation `t.pval=0` specifically?
